x/wasm/simulation/proposals.go

```diff
@@ -0,0 +1,221 @@
+package simulation
+
+import (
+	"math/rand"
+
+	"github.com/CosmWasm/wasmd/app/params"
+	"github.com/CosmWasm/wasmd/x/wasm/keeper/testdata"
+	"github.com/CosmWasm/wasmd/x/wasm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+)
+
+const (
+	WeightStoreCodeProposal           = "weight_store_code_proposal"
+	WeightInstantiateContractProposal = "weight_instantiate_contract_proposal"
+	WeightUpdateAdminProposal         = "weight_update_admin_proposal"
+	WeightExeContractProposal         = "weight_execute_contract_proposal"
+	WeightClearAdminProposal          = "weight_clear_admin_proposal"
+)
+
+func ProposalContents(bk BankKeeper, wasmKeeper WasmKeeper) []simtypes.WeightedProposalContent {
+	return []simtypes.WeightedProposalContent{
+		// simulation.NewWeightedProposalContent(
+		// 	WeightStoreCodeProposal,
+		// 	params.DefaultWeightStoreCodeProposal,
+		// 	SimulateStoreCodeProposal(wasmKeeper),
+		// ),
+		simulation.NewWeightedProposalContent(
+			WeightInstantiateContractProposal,
+			params.DefaultWeightInstantiateContractProposal,
+			SimulateInstantiateContractProposal(
+				bk,
+				wasmKeeper,
+				DefaultSimulationCodeIDSelector,
+			),
+		),
+		simulation.NewWeightedProposalContent(
+			WeightUpdateAdminProposal,
+			params.DefaultWeightUpdateAdminProposal,
+			SimulateUpdateAdminProposal(
+				wasmKeeper,
+				DefaultSimulateUpdateAdminProposalContractSelector,
+			),
+		),
+		simulation.NewWeightedProposalContent(
+			WeightExeContractProposal,
+			params.DefaultWeightExecuteContractProposal,
+			SimulateExecuteContractProposal(
+				bk,
+				wasmKeeper,
+				DefaultSimulationExecuteContractSelector,
+				DefaultSimulationExecuteSenderSelector,
+				DefaultSimulationExecutePayloader,
+			),
+		),
+		simulation.NewWeightedProposalContent(
+			WeightClearAdminProposal,
+			params.DefaultWeightClearAdminProposal,
+			SimulateClearAdminProposal(
+				wasmKeeper,
+				DefaultSimulateClearAdminProposalContractSelector,
+			),
+		),
+	}
+}
+
+// simulate store code proposal (unused now)
+// Current problem: out of gas (defaul gaswanted config of gov SimulateMsgSubmitProposal is 10_000_000)
```

Comment:
Do you have a proposal for a better gas value? We can open an issue/PR on the SDK to get this addressed

Reply:
I think we have 2 solution to solve this problem:
- Open a PR to increase defaul gaswanted so that this proposal can easily pass
- Write a simple .wasm code which size is smaller than current. So that it can decrease the gas used.
But the 2 solution quite difficult. Because this may change all the current sims test. One more thing, in the future, we need to implement sudo entry point for Sudo proposal (it will increase the size of .wasm contract)

Reply:
Store code and sudo gov operations are well covered by integration tests already. Simulations are just for regression and fuzzing. They don't add super much value here IMHO. Let's have a follow up task to revisit the store code when we are on SDK v0.47. There is some work on the gov module.